### DISCUSSION TITLE
Questions about the 'dev-agent.sh' script

### DISCUSSION THREAD

**Discussion by YYM0093** [k3s-io/k3s#11158] [General]
I found two scripts, 'dev-agent.sh' and 'dev-docker-agent.sh', in the /scripts directory. Based on their content, it seems that they compile /cmd/agent/main.go to produce a dedicated binary for the agent, k3s-agent. Compared to /cmd/server/main.go, it appears to have fewer unnecessary components for the agent. In theory, an agent started this way should use significantly less memory. However, after checking the memory composition with the pmap command, I noticed that the static memory of the .text segment has not changed much. I would like to know the true purpose of these two scripts, 'dev-agent.sh' and 'dev-docker-agent.sh'?

![image](https://github.com/user-attachments/assets/e2e5aca8-19a5-42d8-915e-f5ee61454b78)

![image](https://github.com/user-attachments/assets/fdf522ea-c0f5-40fe-af01-f3884615c456)

**Comment by brandond** [comment]
I'm not aware of anyone that uses these any more. They may have been part of the developer workflow for folks that are no longer on the team, or maybe some folks are still using them but if so it's not advertised. You're welcome to use them if you want and they work for you, but I wouldn't read too much into the performance or optimization of whatever they run or build.

**Reply by YYM0093** [reply]
Alright, I understand. Thank you for your reply! If anyone else is trying to use these two script files, feel free to discuss together.
